### DISCUSSION TITLE
`volta pin` write to different file

### DISCUSSION THREAD

**Discussion by jensmeindertsma** [pnpm/pnpm#6651] [Ideas]
I'm using pnpm, and want to write a lean Dockerfile using this guide:
https://pnpm.io/cli/fetch
This requires that `package.json` is not added until after running `pnpm fetch`.
But Volta only installs pnpm when pinned in `package.json`. So can we allow storing the pinned versions in a separate JSON file that can be added instead of package.json?

**Comment by zkochan** [comment]
Isn't that something that volta would have to support?

**Comment by jensmeindertsma** [comment]
Yes. I was thinking `volta.json`

**Reply by weyert** [reply]
You might want to consider `asdf` with the node and pnpm plugins
